### PR TITLE
feat: add support for custom status

### LINKED ISSUE #2204: Support for custom status
### Summary

https://github.com/discord/discord-api-docs/pull/6345

### What is the feature request for?

The core library

### The Problem

API change

### The Ideal Solution

Reflect the API

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Dorukyum**
## Summary
Closes #2204.
Defaults `Activity.name` to `"Custom Status"` if the type is `ActivityType.custom` when initializing, but not sure if this is the best way to do it. This currently requires developers to explicitly set the `type` field to `ActivityType.custom` because it defaults to `ActivityType.unknown`, as seen in the example below:

```py
await bot.change_presence(activity=discord.Activity(type=discord.ActivityType.custom, state="I'm Batman."))
```

I feel like this should be made easier on the developer frontend, resulting with the following:

```py
await bot.change_presence(activity=discord.Activity(state="I'm Batman."))
# or maybe even this:
await bot.change_presence(activity=discord.Activity("I'm Batman."))
```

I'm ready to push the change necessary to implement the last suggestion if the idea is approved.

## Information
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist
- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by Lulalaby — DISMISSED** [review]

**Comment by Dorukyum** [comment]
`CustomActivity.state` will now be set to `.name` by default, which means developers can now write the following code to create a custom bot status:
```py
activity = discord.CustomActivity("I'm Batman.")
```
This will have the same effect as:
```py
activity = discord.Activity(type=discord.ActivityType.custom, state="I'm Batman.")
```

**Comment by Om1609** [comment]
LGTM

File: discord/activity.py [hunk 1]
```diff
@@ -769,7 +774,7 @@ def __init__(
     ):
         super().__init__(**extra)
         self.name: str | None = name
-        self.state: str | None = extra.pop("state", None)
+        self.state: str | None = extra.pop("state", name)
         if self.name == "Custom Status":
             self.name = self.state
```
Comment (Lulalaby) [inline discord/activity.py:779]: sure about that?

Comment (Dorukyum) [reply discord/activity.py:779]: Sure about L777, yeah, but L778-779 is out of the scope of this pr.

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**